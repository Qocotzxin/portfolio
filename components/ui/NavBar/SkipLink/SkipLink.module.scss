.SkipLink {
  padding: var(--chakra-space-1) var(--chakra-space-2);
  width: fit-content;
  background: var(--chakra-colors-teal-500);
  font-size: var(--chakra-fontSizes-sm);
  font-weight: var(--chakra-fontWeights-bold);
  position: absolute;
  left: 0;
  right: 0;
  margin: auto;
  border-radius: var(--chakra-radii-sm);
  transform: translateY(-250%);
  transition: transform 0.3s;
  z-index: var(--chakra-zIndeces-skipLink);

  &:focus {
    transform: translateY(10%);
  }
}
