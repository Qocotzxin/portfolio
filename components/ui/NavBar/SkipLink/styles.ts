import styles from "./SkipLink.module.scss";

export const Classes = {
  base: styles.SkipLink,
};
