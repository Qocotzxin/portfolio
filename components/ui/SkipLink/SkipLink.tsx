import { Link, VisuallyHidden } from "@chakra-ui/react";

const SkipLink = () => {
  return (
    <VisuallyHidden>
      <Link href="#main"></Link>
    </VisuallyHidden>
  );
};

export default SkipLink;
