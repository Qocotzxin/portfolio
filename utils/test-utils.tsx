import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
