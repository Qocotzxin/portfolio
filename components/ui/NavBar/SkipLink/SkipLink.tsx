import { useHygraphContext } from "@contexts/hygraphContext";
import { FC } from "react";
import { Classes } from "./styles";

const SkipLink: FC = () => {
  const { skipLinks } = useHygraphContext();

  return skipLinks && skipLinks.length ? (
    <a data-testid="SkipLink" href="#main" className={Classes.base}>
      {skipLinks[0].text}
    </a>
  ) : null;
};

export default SkipLink;
