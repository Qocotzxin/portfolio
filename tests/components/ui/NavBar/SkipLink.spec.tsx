import { render, screen } from "@utils/test-utils";
import SkipLink from "@ui/NavBar/SkipLink";

describe("SkipLink", () => {
  it("Should render the skip link as long as the skipLinks array is defined and has elements in it, and match the snapshot.", () => {
    const { container } = render(<SkipLink />);
    expect(screen.getByTestId("SkipLink")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
