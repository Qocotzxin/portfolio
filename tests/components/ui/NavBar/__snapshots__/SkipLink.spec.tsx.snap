// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink Should render the skip link as long as the skipLinks array is defined and has elements in it, and match the snapshot. 1`] = `
<div>
  <a
    class="SkipLink"
    data-testid="SkipLink"
    href="#main"
  >
    Skip to content
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
